src/roles/common/tasks/wait-for-ssh-port.yml

```diff
@@ -1,27 +1,34 @@
----
-- name: Wait for connection to be ready
-  wait_for:
-    port: "22"
-    host: "{{ ssh_host }}"
-    search_regex: OpenSSH
-    delay: "{{ ssh_delay_seconds | default(5) }}"
-    timeout: "{{ ssh_timeout_seconds | default(300) }}"
-  remote_user: "{{ ssh_proxy_username | default(None) }}"
-  delegate_to: "{{ ssh_proxy_host | default('localhost') }}"
-
-- name: Add proxy setup
-  set_fact:
-    proxy_conf: '-o ProxyCommand="ssh -W %h:%p -q {{ ssh_proxy_configuration }}"'
-  when: ssh_proxy_configuration is defined
-
-- name: Check connectivity to the target
-  command: >-
-    sshpass -p{{ ssh_password }} ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null
-    {{ proxy_conf | default("") }} -tt {{ ssh_user }}@{{ ssh_host }}
-  register: ssh_result
-  no_log: "{{ lookup('env', 'METROAE_NO_LOG') or 'true' }}"
-  failed_when: not ssh_result.rc == 0
-  when: check_login | default(False)
-  delegate_to: localhost
-  retries: 10
-  delay: "{{ ssh_delay_seconds | default(10) }}"
+---
+- name: Wait for connection to be ready
+  wait_for:
+    port: "22"
+    host: "{{ ssh_host }}"
+    search_regex: OpenSSH
+    delay: "{{ ssh_delay_seconds | default(5) }}"
+    timeout: "{{ ssh_timeout_seconds | default(300) }}"
+  remote_user: "{{ ssh_proxy_username | default(None) }}"
+  delegate_to: "{{ ssh_proxy_host | default('localhost') }}"
+
+- name: Add proxy setup
+  set_fact:
+    proxy_conf: '-o ProxyCommand="ssh -W %h:%p -q {{ ssh_proxy_configuration }}"'
+  when: ssh_proxy_configuration is defined
+
+- name: Get running userinfo for ansible
+  become: false
+  local_action: command whoami
+  register: username_on_the_host
+
+- name: Check connectivity to the target
+  command: >-
+    sshpass -p{{ ssh_password }} ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null
+    {{ proxy_conf | default("") }} -tt {{ ssh_user }}@{{ ssh_host }}
+  register: ssh_result
+  no_log: "{{ lookup('env', 'METROAE_NO_LOG') or 'true' }}"
+  failed_when: not ssh_result.rc == 0
+  when: check_login | default(False)
+  delegate_to: localhost
+  retries: 10
+  delay: "{{ ssh_delay_seconds | default(10) }}"
+  remote_user: "{{ username_on_the_host }}"
```

Comment:
Why was this needed?

Reply:
Because of Ansible flakiness. In the portal deployment I have to start the vm and wait for it start before I can login and grow the partition. But the place I am calling it from has remote_user defined as target server username which is being carried over into this file so I have to override it here.

Reply:
These wait for ssh calls are all over the place.  I don't want to pollute the common code because of a weird special case with just this.  Can you put the logic in the portal code where it is called and override remote_user?

Reply:
I understand the code is all over the place. but if you notice from places it is called everything by default was using the username for the Ansible host. I am just making it explicit by doing this.

Reply:
Ok, this prevents the issue from happening elsewhere.